mergify_engine/rules/__init__.py

```diff
@@ -425,16 +474,46 @@ def get_annotations(self, path):
 class MergifyConfig(typing.TypedDict):
     pull_request_rules: PullRequestRules
     queue_rules: QueueRules
+    defaults: Defaults
+
+
+def merge_config(config: typing.Any) -> typing.Any:
```

Comment:
does not look like it's really `config: Any` since you use it as a `dict` just after. Don't lie! 😆